### DISCUSSION TITLE
Improve BatchGet / BatchWrite to iterate over 25 records.

### DISCUSSION THREAD

**Discussion by ebisbe** [sensedeep/dynamodb-onetable#235] [Ideas]
I have a batch process that does 3 simple things. Query items ( I've limited to 500 ) from DDB, each item is sent to a queue and then deletes each item. 
I'd like to be able to do something like:

```
const Items = Model.find(..., {limit: 500})
let batch = {}
for(const Item of Items){
    // send sqs message
   model.remove(..., {batch})
}
table.batchWrite(batch)
```

But right now I need to do with batchWrite:

```
const Items = Model.find(..., {limit: 500})
const operations = []
for(const Item of Items){
    // send sqs message
   operations.push({...})
}
do {
    let batch = {}
    const operationsSlice = operations.slice(0, 24)
    operationsSlice.forEach(async operation => await model.remove(operation, { batch }))
    await table.batchWrite(batch)
} while (actions > 0)
```

It could be real handy that the batchGet/Write handles the limit imposed by DDB to 25 items by doing the slicing himself. Maybe add an option so the user acknowledges he is sending more than 25 items.

**Comment by mobsense** [comment]
That is a good idea.

I'll convert this to an issue.
